### PR TITLE
permit symlinks in config directory

### PR DESCRIPTION
**Pull request by roobert**
We have a situation where our `conf.d` directory is maintained by Puppet, but our application layer is deployed by Ansible. We deploy apps into `opt/<app>-<instance>` and the sensu check configs into a subdir called `checks`. We then create a symlink `/etc/sensu/conf.d/app/<app>-<id>` to the `checks` dir so that if an app gets removed we're just left with a dead symlink, rather than Sensu loading checks for apps that have been removed. Later, a puppet job comes along and removes dead symlinks.

Convoluted, I know. Regardless, I think permitting symlinks in `conf.d` should be acceptable :)

### CODE REVIEW

**Comment by portertech** [comment]
You can specify multi configuration directories in `/etc/default/sensu` w/ `CONFIG_DIR=/some/dir,/another/dir`.

**Comment by roobert** [comment]
@portertech: this ok?

File: spec/loader_spec.rb [hunk 1]
```diff
@@ -139,7 +139,7 @@
   it "can load settings from files in a directory" do
     @loader.load_directory(@config_dir)
     warnings = @loader.warnings
-    expect(warnings.size).to eq(4)
+    expect(warnings.size).to eq(6)
```
Comment (portertech) [inline spec/loader_spec.rb]: Why did this increase by 2?

Comment (portertech) [inline spec/loader_spec.rb]: "loading config file" & "config file applied changes"

File: lib/sensu/settings/loader.rb [hunk 2]
```diff
@@ -124,7 +124,7 @@ def load_file(file)
       def load_directory(directory)
         warning("loading config files from directory", :directory => directory)
         path = directory.gsub(/\\(?=\S)/, "/")
-        Dir.glob(File.join(path, "**/*.json")).each do |file|
+        Dir.glob(File.join(path, "**{,/*/**}/*.json")).uniq.each do |file|
```
Comment (portertech) [inline lib/sensu/settings/loader.rb]: What's the uniq() for? Shouldn't have duplicate file paths, no?

Comment (roobert) [inline lib/sensu/settings/loader.rb]: The glob seems to match files multiple times:

```
[7] pry(main)> Dir.glob(File.join("**/*.json")) { |i| puts i }
nested/file.json
merger.json
=> nil
[8] pry(main)> Dir.glob(File.join("**{,/*/**}/*.json")) { |i| puts i }
nested/file.json
merger.json
nested/file.json
alternative/symlinked.json
=> nil
```

Took the glob from here: http://stackoverflow.com/questions/357754/can-i-traverse-symlinked-directories-in-ruby-with-a-glob

**Comment by roobert** [comment]
gif

File: lib/sensu/settings/loader.rb (hunk 2 above)
Comment (portertech) [inline lib/sensu/settings/loader.rb]: Seems we need a `uniq()` anyways, as this pattern along w/ alternatives, will pick up a symlink to a dir that it has already listed :+1:

Comment (portertech) [inline lib/sensu/settings/loader.rb]: Although, it is still a uniq file path :P

**Pull request merged**